src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java

```diff
@@ -56,17 +59,38 @@ private String addExtension(String name, String [] extensions) {
 		return name + "." + extensions[0];
 	}
 	
+	boolean detailGCodeHeader = true;
+	
 	private void save(String filename, Turtle turtle, Plotter robot) throws Exception {
 		logger.debug("saving...");
 		
 		try (Writer out = new OutputStreamWriter(new FileOutputStream(filename))) {
+			
+			if (detailGCodeHeader) {
+				out.write(";FLAVOR:Marlin-polargraph\n");
+
+				//out.write(";TIME:"+ "\n"); // TODO
+				
+				Rectangle2D.Double bounds = turtle.getBounds();
+				out.write(";MINX:" + StringHelper.formatDouble(bounds.x) + "\n");
+				out.write(";MINY:" + StringHelper.formatDouble(bounds.y) + "\n");
+				//out.write(";MINZ:0.000\n");
+				out.write(";MAXX:" + StringHelper.formatDouble(bounds.width + bounds.x) + "\n");
+				out.write(";MAXY:" + StringHelper.formatDouble(bounds.height + bounds.y) + "\n");
+				//out.write(";MAXZ:0.000\n");
 
+				out.write(";Generated with " + MakelangeloVersion.getFullOrLiteVersionStringRelativeToSysEnvDevValue() + "\n");
+			}
+			
 			SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd 'at' HH:mm:ss z");
 			Date date = new Date(System.currentTimeMillis());
 			out.write("; " + formatter.format(date) + "\n");
+			
 			// TODO MarlinPlotterInterface.getFindHomeString()?
 			out.write("G28\n");  // go home
 
+			// TODO user "start gcode"
```

Comment:
If your PR is ready to be merged, there should be no TODO. If you want to keep track of something else to do, please create an issue.

Reply:
Issue created. thank you.
But I would have kept the TODO in the code too ... (double reminder )